Geometry_Engine/Query/GeometryHash.cs

```diff
@@ -0,0 +1,409 @@
+                foreach (var faceIndex in obj.Faces[i].FaceIndices())
+                {
+                    if (dic.ContainsKey(faceIndex))
+                        dic[faceIndex] += i;
+                    else
+                        dic[faceIndex] = i;
+                }
+            }
+
+            List<double> result = new List<double>();
+
+            for (int i = 0; i < obj.Vertices.Count; i++)
+            {
+                int pointTranslationFactor;
+                if (!dic.TryGetValue(i, out pointTranslationFactor))
+                    pointTranslationFactor = 0;
+
+                result.AddRange(obj.Vertices[i].ToDoubleArray(pointTranslationFactor + translationFactor));
+            }
+
+            return result.ToArray();
+        }
+
+
+        /***************************************************/
+        /****  Vector                                   ****/
+        /***************************************************/
+
+        [Description("The GeometryHash for a Point is simply an array of 3 numbers composed by the Point X, Y and Z coordinates.")]
+        private static double[] GeometryHash(this Point obj, double translationFactor)
+        {
+            return obj.ToDoubleArray(translationFactor);
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for a Plane is the GeometryHash of its Origin point translated by its Normal vector.")]
+        private static double[] GeometryHash(this Plane obj, double translationFactor)
+        {
+            translationFactor += (int)TypeTranslationFactor.Plane;
+
+            return obj.Origin.Translate(obj.Normal).ToDoubleArray(translationFactor);
+        }
+
+
+        /***************************************************/
+        /****  Other methods                            ****/
+        /***************************************************/
+
+        // Fallback
+        private static double[] GeometryHash(this object obj, double translationFactor)
+        {
+            BH.Engine.Base.Compute.RecordError($"Could not find a {nameof(GeometryHash)} method for type {obj.GetType().FullName}.");
+            return new double[] { };
+        }
+
+        /***************************************************/
+
+        private static double[] ToDoubleArray(this Point p, double typeTranslationFactor, double[] translationArray = null)
+        {
+            return new double[]
+            {
+                p.X + typeTranslationFactor + translationArray?.ElementAtOrDefault(0) ?? 0,
+                p.Y + typeTranslationFactor + translationArray?.ElementAtOrDefault(1) ?? 0,
+                p.Z + typeTranslationFactor + translationArray?.ElementAtOrDefault(2) ?? 0
+            };
+        }
+
+        /***************************************************/
+
+        private static double[] ToDoubleArray(this IEnumerable<Point> points, double typeTranslationFactor, double[] translationArray = null)
```

Comment:
maybe no hard in keeping really, but do we use/need this method?

Reply:
Yes it's used on line 219 - thought I'd use it more but no harm in keeping it I think